dark/reads.py

```diff
@@ -1126,6 +1129,10 @@ def filter(self, read):
                 (self.maxLength is not None and readLen > self.maxLength)):
             return False
 
+        nFraction = read.sequence.count('N') / readLen
+        if self.maxNFraction is not None and self.maxNFraction < nFraction:
```

Comment:
You should only calculate the N fraction if you actually need it:

```suggestion
        if self.maxNFraction is not None and self.maxNFraction < read.sequence.count('N') / readLen:
```